### ISSUE TITLE
Error occurred when import dolfin module

### ISSUE THREAD

**Issue by zhuliwuhan** [Juanlu001/fenics-recipes#24]
Hi,
My OS is ubuntu 14.04. I am using Anaconda Python distribution.
I run the line "conda i nstall fenics mkl --channel juanlu001" to install fenics

But something went wrong as I import dolfin. Error information as follow:
'' libboost_filesystem.so.1.55.0: cannot open shared object file: No such file or directory ''

I found that there is '' libboost_filesystem.so.1.57.0"  file in my
computer. How can I deal with that?

Thanks 

Li

**Comment by GaZ3ll3** [comment]
possibly duplicate here with #8 , you can install libboost 1.55 on your own.

```
conda install "boost=1.55" --channel juanlu001
```

however, I encountered this before on 14.04, the `libboost` problem was gone , but `libcppunit` problem will come out, use `apt-get libcppunit-1.13.0` can fix, while finally, a problem related to `libstdc++` will come up.

```
ImportError: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.8' not found (required by /home/lurker/Documents/anaconda/envs/fenics/lib/python2.7/site-packages/dolfin/cpp/_common.so
```

which indicates 

> If you already have an older version of libstdc++ installed then the error might look like one of the following instead:

```
./a.out: /usr/lib/libstdc++.so.6: version `GLIBCXX_3.4.20' not found
./a.out: /usr/lib/libstdc++.so.6: version `CXXABI_1.3.8' not found
```

> This means the linker found /usr/lib/libstdc++.so.6 but that library belongs to an older version of GCC than was used to compile and link the program a.out (or some part of it). The program depends on code defined in the newer libstdc++ that belongs to the newer version of GCC, so the linker must be told how to find the newer libstdc++ shared library. 

which means `gcc-4.8` is kind of old for building. Probably installing `gcc-4.9` will help. See here
http://askubuntu.com/questions/466651/how-do-i-use-the-latest-gcc-4-9-on-ubuntu-14-04

**Comment by astrojuanlu** [comment]
Thanks @zhuliwuhan for the bug report and @GaZ3ll3 for the detailed walkthrough. I am trying to reproduce your steps but I am facing an unrelated error.

On the other hand, unless there is no other way I will try to use all the default packages in Ubuntu 14.04 for building.

**Comment by zhuliwuhan** [comment]
Thanks @GaZ3ll3  and @Juanlu001 . I  installed libboost 1.55 as @GaZ3ll3  mentioned,  however another problem related to "libblas.so" come up. So I am still stuck in there. 

I went back to installation webpage on Fenicsd Project. There is another way to install by using Ubuntu PPA. So I added the FEniCS PPA to my system to get  the latest FEniCS release.  See here http://fenicsproject.org/download/ubuntu_details.html

It works quite good and simply. I think it's an alternative way for me. 
Thanks you all the same.

**Comment by astrojuanlu** [comment]
Exactly, I can't find "libblas.so" either. That is what I am trying to fix now.

**Comment by astrojuanlu** [comment]
I uploaded a new dolfin package which should fix the libblas linking problem and the boost version specification. Could you please confirm if it works? Do `conda install dolfin --channel juanlu001`.

**Comment by GaZ3ll3** [comment]
I kinda know why mine works, I have already MKL installed and configured the path, and I looked at the installation log, it seems BLAS dir is already detected somehow.

**Comment by astrojuanlu** [comment]
I have tried the new packages in a couple of different distros and everything seems to work. I am closing this issue, please don't hesitate to reopen if it persists or open a new one.

**Comment by zhuliwuhan** [comment]
Thank you @Juanlu001 and @GaZ3ll3. Everything works fine now after updating the dolfin package.

**Comment by ghost** [comment]
Hi @Juanlu001, I am trying to install FEniCS via your channel. Upon running the demo `~/.anaconda3/envs/fenics27/share/dolfin/demo/documented/poisson/python/demo_poisson.py` I get an error related to `boost`:

```
ImportError:/home/user/.anaconda3/envs/fenics27/lib/python2.7/site-packages/ufc/../../../libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /usr/lib/libboost_filesystem.so.1.58.0);
```

Apparently FEniCS/Dolfin is trying to use the system version of boost instead of the one from your channel.

ps: I am on Arch Linux and I am using the Python 3.x version of Anaconda. The latter should not matter, I think, because FEniCS is installed in a separate environment with Python 2.x

**Comment by astrojuanlu** [comment]
Hello @cfbaptista, that is strange. Can you try the following and post the outputs somewhere? (Maybe gist.github.com)

``` bash
$ source activate fenics27
$ conda info -a
$ conda list
$ which python
$ python ~/.anaconda3/envs/fenics27/share/dolfin/demo/documented/poisson/python/demo_poisson.py
```

**Comment by Juanlu001** [comment]
Hello @cfbaptista, that is strange. Can you try the following and post the outputs somewhere? (Maybe gist.github.com)

```bash
$ source activate fenics27
$ conda info -a
$ conda list
$ which python
$ python ~/.anaconda3/envs/fenics27/share/dolfin/demo/documented/poisson/python/demo_poisson.py
```

**Comment by ghost** [comment]
Hi @Juanlu001, I have posted the output you requested at [this](https://gist.github.com/cfbaptista/37d95323b8a64623ee08#file-fenics_terminal_output) gist. Thanks for your support!

**Comment by astrojuanlu** [comment]
I opened a new issue to discuss that, please head there. Thanks!

**Comment by mthewissen** [comment]
I don't think this should be closed. I still encounter exactly the same error as the OP (be it on CentOS 6.6). Trying the `demo_poisson.py` also gives

```
ImportError: libboost_filesystem.so.1.55.0: cannot open shared object file: No such file or directory
```

Upon installation it says indeed:

```
The following NEW packages will be INSTALLED:
    boost:            1.57.0-4             defaults       
    ect.
```

Whereas this ought to be 1.55 for dolfin version 1.4.
Can you revert to a latest working version of some kind? Thanks for maintaining.

**Comment by mthewissen** [comment]
FWIW, I managed to get version 1.4 working on an older system using the following:

```
conda install dolfin=1.4.0 boost=1.55 eigen3 ffc petsc petsc4py instant ufl fiat swig=3.0.2 vtk=5.10 fenics=1.4.0 mkl scientificpython --channel https://conda.binstar.org/juanlu001/channel/fenics:1.4.0:centos
```

**Comment by GitH00t** [comment]
Hello,

I am experiencing a similar issue as OP. 
I followed the standard recipe using conda.
Ubuntu 15.10

```
ImportError: libgfortran.so.1: cannot open shared object file: No such file or directory
```


I posted the results of 
```
$ conda info -a
$ conda list
$ which python
$ python ~/.anaconda3/envs/fenics27/share/dolfin/demo/documented/poisson/python/demo_poisson.py
```
[https://gist.github.com/GitH00t/9062fdb3122fa20ceab0](url)

When I upgrade numpy I can get rid of the problem, but the I get the same error as OP

```
ImportError: libboost_system.so.1.57.0: cannot open shared object file: No such file or directory
```

Since you pointed out in the faq that this might be related to linking I executed the command you specified and posted it to this gist

[https://gist.github.com/GitH00t/c7f9c9bf7f3aab97e28a](url)

Unfortunately I am pretty newb at fenics and linux, so I don't really know what this is telling me.
